src/analytics/useAnalytics.ts

```diff
@@ -0,0 +1,8 @@
+import { useCallback } from "react";
```

Comment:
Given the dependency is [], so you do not want to evaluate again, since you already define the function outside of the function, you can directly import the function instead of create the hook, which basically the same and less complicated